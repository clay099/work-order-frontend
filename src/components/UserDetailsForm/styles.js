import { makeStyles } from "@material-ui/core/styles";
import { form, submit } from "../../styles/styleObjects";

const useStyles = makeStyles((theme) => {
	let formObj = form(theme);
	let submitObj = submit(theme);
	return { ...formObj, ...submitObj };
});

export default useStyles;
