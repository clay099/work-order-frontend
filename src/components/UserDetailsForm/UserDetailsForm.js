import React from "react";
import { Button, TextField, Grid } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import FormAddressItems from "../FormAddressItems/FormAddressItems";

const useStyles = makeStyles((theme) => ({
	form: {
		width: "100%", // Fix IE 11 issue.
		marginTop: theme.spacing(3),
	},
	submit: {
		margin: theme.spacing(3, 0, 2),
	},
}));

export default function UserDetailsForm({
	userType,
	handleSubmit,
	formData,
	handleChange,
}) {
	const classes = useStyles();

	return (
		<div>
			<form className={classes.form} noValidate>
				<Grid container spacing={2}>
					<Grid item xs={12} sm={6}>
						<TextField
							name="firstName"
							variant="outlined"
							required
							fullWidth
							id="firstName"
							label="First Name"
							autoFocus
							onChange={handleChange}
							value={formData.firstName}
						/>
					</Grid>
					<Grid item xs={12} sm={6}>
						<TextField
							variant="outlined"
							required
							fullWidth
							id="lastName"
							label="Last Name"
							name="lastName"
							onChange={handleChange}
							value={formData.lastName}
						/>
					</Grid>
					<Grid item xs={12} sm={6}>
						<TextField
							variant="outlined"
							required
							fullWidth
							id="email"
							label="Email Address"
							name="email"
							onChange={handleChange}
							value={formData.email}
						/>
					</Grid>
					<Grid item xs={12} sm={6}>
						<TextField
							variant="outlined"
							required
							fullWidth
							id="phone"
							label="Phone Number"
							name="phone"
							onChange={handleChange}
							value={formData.phone}
						/>
					</Grid>
					{userType === "user" ? (
						<FormAddressItems
							formData={formData}
							handleChange={handleChange}
						/>
					) : null}
					<Grid item xs={12}>
						<TextField
							variant="outlined"
							required
							fullWidth
							name="password"
							label="Password"
							type="password"
							id="password"
							onChange={handleChange}
							value={formData.password}
						/>
					</Grid>
				</Grid>
				<Button
					type="submit"
					fullWidth
					variant="contained"
					color="primary"
					className={classes.submit}
					onClick={handleSubmit}
				>
					Sign Up New {userType}
				</Button>
			</form>
		</div>
	);
}
