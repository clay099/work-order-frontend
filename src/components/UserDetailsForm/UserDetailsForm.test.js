import React from "react";
import UserDetailsForm from "./UserDetailsForm";
import {
	render,
	userTypeUserState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
} from "../../testHelper/testHelper";
console.error = jest.fn();

describe("UserDetailsForm Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(
			<UserDetailsForm
				userType={userTypeUserState}
				handleSubmit={handleSubmitState}
				handleChange={handleChangeState}
				disabled={disabledState}
				disabledEmail={disabledEmailState}
				buttonText={buttonTextState}
				formData={formDataState}
			/>
		);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(
			<UserDetailsForm
				userType={userTypeUserState}
				handleSubmit={handleSubmitState}
				handleChange={handleChangeState}
				disabled={disabledState}
				disabledEmail={disabledEmailState}
				buttonText={buttonTextState}
				formData={formDataState}
			/>
		);
		expect(asFragment).toMatchSnapshot();
	});

	test("last name to be in the document", () => {
		const { getByText } = render(
			<UserDetailsForm
				userType={userTypeUserState}
				handleSubmit={handleSubmitState}
				handleChange={handleChangeState}
				disabled={disabledState}
				disabledEmail={disabledEmailState}
				buttonText={buttonTextState}
				formData={formDataState}
			/>
		);
		const value = getByText("Last Name");
		expect(value).toBeInTheDocument();
	});
});
