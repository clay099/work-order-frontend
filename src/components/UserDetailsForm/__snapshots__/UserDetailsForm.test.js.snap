// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailsForm Component matches snapshot 1`] = `[Function]`;
